app/src/main/java/com/umc/edison/presentation/login/IdentityTestViewModel.kt

```diff
@@ -73,94 +63,99 @@ class IdentityTestViewModel @Inject constructor(
-                it.copy(
-                    identity = it.identity.copy(
-                        selectedKeywords = it.identity.selectedKeywords - keyword
-                    )
-                )
-            }
+        val state = uiState.value
+        val category = state.currentCategory
+        val currentIdentity = state.identities[category] ?: IdentityModel.DEFAULT
+        val selected = currentIdentity.selectedKeywords
+
+        val updated = if (selected.contains(keyword)) {
+            selected - keyword
         } else {
-            if (uiState.value.identity.selectedKeywords.size >= 5) {
+            if (selected.size >= 5) {
                 showToast("최대 5개의 키워드를 선택할 수 있습니다.")
-            } else {
-                _uiState.update {
-                    it.copy(
-                        identity = it.identity.copy(
-                            selectedKeywords = it.identity.selectedKeywords + keyword
-                        )
-                    )
-                }
+                return
             }
+            selected + keyword
         }
-    }
 
-    // TODO: Implement this function
-    fun toggleInterestKeyword(keyword: KeywordModel) {
-//        if (uiState.value.identity.selectedKeywords.contains(keyword)) {
-//            _uiState.update {
-//                it.copy(
-//                    identity = it.identity.copy(
-//                        selectedKeywords = it.identity.selectedKeywords - keyword
-//                    )
-//                )
-//            }
-//        } else {
-//            if (uiState.value.identity.selectedKeywords.size >= 5) {
-//                _baseState.update {
-//                    it.copy(toastMessage = "최대 5개의 키워드를 선택할 수 있습니다.")
-//                }
-//            } else {
-//                _uiState.update {
-//                    it.copy(
-//                        interest = it.interest.copy(
-//                            selectedKeywords = it.interest.selectedKeywords + keyword
-//                        )
-//                    )
-//                }
-//            }
-//        }
-    }
-
-    // TODO: Implement this function
-    fun setInterestTestResult(navController: NavHostController) {
-//        collectDataResource(
-//            flow = setUserInterestUseCase(_uiState.value.interest.toDomain()),
-//            onSuccess = {
-//                _uiState.update { it.copy(interest = it.interest.copy(options = it.interest.selectedKeywords)) }
-//                CoroutineScope(Dispatchers.Main).launch {
-//                    navController.navigate(NavRoute.MyEdison.route)
-//                }
-//            },
-//            onError = { error ->
-//                _baseState.update { it.copy(toastMessage = "$error") }
-//            },
-//        )
+        _uiState.update {
+            it.copy(
+                identities = it.identities.toMutableMap().apply {
+                    put(category, currentIdentity.copy(selectedKeywords = updated))
+                }
```

Comment:
여기도 마찬가지로 아래처럼 수정 가능해요!

```suggestion
                identities = it.identities + (category to currentIdentity.copy(selectedKeywords = updated))
                }
```